src/core/project/qgsprojectviewsettings.h

```diff
@@ -104,6 +104,20 @@ class CORE_EXPORT QgsProjectViewSettings : public QObject
      */
     void setPresetFullExtent( const QgsReferencedRectangle &extent );
 
+    /**
+     * Sets whether the project's preset full extent should be restored when the project is loaded.
+     *
+     * \since QGIS 3.44
+     */
+    void setRestoreProjectExtentOnProjectLoad( bool state );
+
+    /**
+     * Returns whether the project's preset full extent should be restored when the project is loaded.
+     *
+     * \since QGIS 3.44
+     */
+    bool mainCanvasOpensAtProjectExtent();
```

Comment:
That name needs to be updated to match the setter function naming (which is an improvement, thanks for that!)